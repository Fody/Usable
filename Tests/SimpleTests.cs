using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

[TestFixture]
[UseReporter(typeof(DiffReporter))]
public class SimpleTests
{
    [Test]
    public void SingleDisposable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleCases::SingleDisposable"));
    }

    [Test]
    public void AlreadyUsing()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleCases::AlreadyUsing"));
    }

    [Test]
    public void VariableReuse()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleCases::VariableReuse"));
    }

    [Test]
    public void NestedUsings()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleCases::NestedUsings"));
    }

    [Test]
    public void MultipleUsings()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleCases::MultipleUsings"));
    }
}