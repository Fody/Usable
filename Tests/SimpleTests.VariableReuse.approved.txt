.class public auto ansi beforefieldinit SimpleCases
       extends [mscorlib]System.Object
{
  .method public hidebysig instance void 
          VariableReuse() cil managed
  {
    // Code size       50 (0x32)
    .maxstack  2
    .locals init ([0] class [mscorlib]System.IO.StreamWriter w)
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    .line 20,20 : 5,6 'AssemblyToProcess\\SimpleCases.cs'
    IL_0000:  nop
    .line 21,21 : 9,44 ''
    IL_0001:  ldstr      "log.txt"
    IL_0006:  call       class [mscorlib]System.IO.StreamWriter [mscorlib]System.IO.File::CreateText(string)
    IL_000b:  stloc.0
    .line 22,22 : 9,53 ''
    IL_000c:  ldloc.0
    IL_000d:  ldstr      "I'm a lumberjack an' I'm ok."
    IL_0012:  callvirt   instance void [mscorlib]System.IO.TextWriter::WriteLine(string)
    IL_0017:  nop
    .line 23,23 : 9,41 ''
    IL_0018:  ldstr      "log2.txt"
    IL_001d:  call       class [mscorlib]System.IO.StreamWriter [mscorlib]System.IO.File::CreateText(string)
    IL_0022:  stloc.0
    .line 24,24 : 9,53 ''
    IL_0023:  ldloc.0
    IL_0024:  ldstr      "I'm a lumberjack an' I'm ok."
    IL_0029:  callvirt   instance void [mscorlib]System.IO.TextWriter::WriteLine(string)
    IL_002e:  nop
    IL_002f:  ret
  } // end of method SimpleCases::VariableReuse
} // end of class SimpleCases